src/commands/showReferrer.ts

```diff
@@ -0,0 +1,88 @@
+import * as vscode from 'vscode';
+import { IActionContext } from '@microsoft/vscode-azext-utils';
+import { execAsync } from '../support/execAsync';
+import { window } from 'vscode';
+import * as os from 'os';
+import * as path from 'path';
+import { access } from 'fs/promises'; 
+
+const errorMessage = 'ORAS not found on default path. Download ORAS or update path: ';
+const goToOrasButton = 'Download ORAS';
+const laterButton = 'Later';
+const validTag = /^[a-z0-9./_:-]+$/; 
+
+// check oras exist on user computer
+async function getOrasPath(path: any, bool : boolean): Promise<[string, boolean]> {
+    const isWindows = os.platform() === 'win32'; 
+    const homedir = os.homedir(); 
+    const orasExecutable = isWindows ? 'oras.exe' : 'oras'; 
+    const dirOras = path.join(homedir, 'bin', 'oras', orasExecutable);
+
+        try {
+            await access(dirOras);
+            return [dirOras , true];
+        } catch {
+            vscode.window.showInformationMessage(errorMessage, goToOrasButton, laterButton)
+            .then(selection => {
+                if (selection === goToOrasButton) {
+                    vscode.env.openExternal(vscode.Uri.parse('https://oras.land/docs/installation'));
+                } else if (selection === laterButton){
+                   //when they choose later it closes the toast 
+                }
+            });
+            return [dirOras, false];
+        } 
+}
+/**
+ * an interface that builds the method getDockerCLiCredentials and returns object type any
+ * we need this to mimic the Docker extensions logInToDockerCli code : node.getDockerCliCredentials because we
+ * can't get the function itself through the instance of the docker extension.
+ */
+interface IAuthentication {
+    getDockerCliCredentials : () => any; 
+}
+
+/**
+ * showReferrer logs into docker cli, calls getOrasPath, checks that an image tag has only the valid characters it needs, 
+ * sends the oras discover command to be executed, and has error handling
+ * @param imageTag 
```

Comment:
If you include @param values in a function comment, you should provide extra info on the comment about them. @param is only useful to specify more info